app/plugin_api_test.go

```diff
@@ -556,6 +556,36 @@ func TestPluginAPIUserCustomStatus(t *testing.T) {
 	userCs, err = th.App.GetCustomStatus(user1.Id)
 	assert.Nil(t, err)
 	assert.Equal(t, csClear, userCs)
+
+	// Set a custom status for clearing it later
+	custom.Emoji = ":100"
+	custom.Text = "happy 100"
+
+	err = api.UpdateUserCustomStatus(user1.Id, custom)
+	assert.Nil(t, err)
+	userCs, err = th.App.GetCustomStatus(user1.Id)
+	assert.Nil(t, err)
+	assert.Equal(t, custom, userCs)
```

Comment:
Can we put a call to `RestoreToPreviousCustomStatus` here, and assert that the status does not change, due to the most recent change having `SetByProduct: false`?

Reply:
yes added